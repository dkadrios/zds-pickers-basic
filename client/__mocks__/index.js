export inputControls from './inputControls'
export mappings from './mappings'
export midi from './midi'
export shiftGroups from './shiftGroups'
export shifter from './shifter'
export store from './store'
export user from './user'
export version from './version'
