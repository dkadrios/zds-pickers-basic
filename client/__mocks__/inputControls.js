export default [
  {
    controlId: 0,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 11,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 1,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 1,
    status: 11,
    value: 111,
    active: false,
    lit: false,
  },
  {
    controlId: 2,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
  {
    controlId: 3,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 4,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 5,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
]
