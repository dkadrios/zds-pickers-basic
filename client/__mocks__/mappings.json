{
  "dialogVisible": false,
  "userDialogVisible": false,
  "userImportDialogVisible": false,
  "channels": [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    "BFD3",
    null,
    null,
    null,
    null,
    null,
    null
  ],
  "stockMappings": [
    "2box",
    "Addictive Drums",
    "Alesis DM Pro",
    "Alesis DM10",
    "Alesis DM5",
    "BFD2",
    "BFD3",
    "Battery 3",
    "E-MU Protean Acoustic",
    "E-MU Protean Bread n Butter",
    "E-MU Protean Studio",
    "E-MU Protean Synth",
    "EZ Drummer The Classic EZX",
    "EZDrummer",
    "EZDrummer Jazz Kit",
    "EZDrummer Latin Percussion",
    "General MIDI",
    "Infinite Instruments",
    "John Emrich - Articulations 1",
    "Kontakt 2",
    "Ocean Way Drums",
    "Pearl Redbox",
    "Roland HandSonic 15",
    "Roland TD-10",
    "Roland TD-10 Perc 1",
    "Roland TD-10 Perc 2",
    "Roland TD-10 Perc 3",
    "Roland TD-10 Perc 4",
    "Roland TD-12",
    "Roland TD-12 Perc 1 (Latin)",
    "Roland TD-12 Perc 2 (Indian)",
    "Roland TD-12 Perc 3 (African)",
    "Roland TD-12 Perc 4 (Salsa)",
    "Roland TD-12 Perc 5 (SFX)",
    "Roland TD-12 Perc 6-8 (User)",
    "Roland TD-20",
    "Roland TD-20 Perc",
    "Roland TD-3",
    "Roland TD-30",
    "Roland TD-30 Perc 1 (Latin)",
    "Roland TD-30 Perc 2 (Indian)",
    "Roland TD-30 Perc 3 (African)",
    "Roland TD-30 Perc 4 (Salsa)",
    "Roland TD-30 Perc 5 (SFX)",
    "Roland TD-30 Perc 6-8 (User)",
    "Roland TD-5",
    "Roland TD-6",
    "Roland TD-6 Perc 1",
    "Roland TD-6 Perc 10 (Special)",
    "Roland TD-6 Perc 2",
    "Roland TD-6 Perc 3 (Room)",
    "Roland TD-6 Perc 4 (Power)",
    "Roland TD-6 Perc 5 (Elec)",
    "Roland TD-6 Perc 6 (808)",
    "Roland TD-6 Perc 7 (Jazz)",
    "Roland TD-6 Perc 8 (Brush)",
    "Roland TD-6 Perc 9 (Perc)",
    "Roland TD-7",
    "Roland TD-8",
    "Roland TD-8 Perc 1",
    "Roland TD-8 Perc 10 (Special)",
    "Roland TD-8 Perc 2",
    "Roland TD-8 Perc 3 (Room)",
    "Roland TD-8 Perc 4 (Power)",
    "Roland TD-8 Perc 5 (Elec)",
    "Roland TD-8 Perc 6 (808)",
    "Roland TD-8 Perc 7 (Jazz)",
    "Roland TD-8 Perc 8 (Brush)",
    "Roland TD-8 Perc 9 (Perc)",
    "Roland TD-9",
    "Roland TD-9 Perc",
    "Simmons SD1000",
    "Simmons SD5X",
    "Simmons SD7",
    "Simmons SD7PK",
    "Simmons SD9K",
    "Sonic Reality iMap Extended",
    "Sonic Reality iMap Standard",
    "Studio ProFiles",
    "Superior Drummer",
    "Yamaha DTX 700",
    "Yamaha DTXplorer",
    "Zendrum Stompblock",
    "Zendrum TRS",
    "Zendrum TRS 2.0"
  ],
  "userMappings": ["BFD3 Custom ZenEdit", "reaper"]
}
