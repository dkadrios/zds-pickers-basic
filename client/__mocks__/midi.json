{
  "devices": [
    {
      "id": "-1325431060",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "output",
      "version": "",
      "state": "connected",
      "connection": "open"
    },
    {
      "id": "-312145282",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "input",
      "version": "",
      "state": "connected",
      "connection": "open"
    }
  ],
  "listeningDevices": ["-312145282"]
}
