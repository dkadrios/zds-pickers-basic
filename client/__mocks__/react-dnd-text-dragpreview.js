/* eslint-disable import/prefer-default-export, jsx-a11y/alt-text */
import React from 'react'

export const createDragPreview = () => <img src={null} />
