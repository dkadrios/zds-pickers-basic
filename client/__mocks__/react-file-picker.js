/* eslint-disable import/prefer-default-export */
export const FilePicker = () => <div>content</div>
