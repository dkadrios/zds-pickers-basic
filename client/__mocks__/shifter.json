{
  "accessGranted": true,
  "errorMessage": "",
  "errorVisible": false,
  "exportBuffer": [],
  "exportDialogVisible": false,
  "exportFilename": "zds-shifter-backup.txt",
  "found": true,
  "hardwareTestVisible": true,
  "importDialogVisible": false,
  "importInProcess": false,
  "invalidSettingsFile": null,
  "midiActivityLEDMode": 0,
  "midiInActivity": false,
  "midiOutActivity": false,
  "ready": true,
  "resetInProcess": false,
  "responding": true,
  "searchedForShifter": true,
  "serialMidiOutEnabled": true,
  "showResetDialog": false,
  "testInterfaceFound": false,
  "usbMidiOutEnabled": true
}
