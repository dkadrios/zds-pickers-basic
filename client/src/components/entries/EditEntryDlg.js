import React, { Component, createRef } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import Button from '@material-ui/core/Button'
import ReactModal from 'react-modal'
import EntryControls from '../controls/EntryControls'
import { actions as shiftGroupActions } from '../../reducers/shiftGroups'
import { actions as shiftEntryActions } from '../../reducers/shiftEntry'
import { instructions } from '../../styles/entryDlg.scss'
import { groupsShape, mappingsShape } from '../../shapes'

/**
 * We ran into a huge issue with material-ui dialog here.
 * When we place components that are in our external zds-pickers library into
 * a modal, we get a stack overflow when the control is focused.
 * This seems to have something to be with there essentially being two separate
 * material-ui instances.
 * To get around this, we're using react-modal instead of the material-ui dialog.
 */

const modalStyle = {
  overlay: {
    backgroundColor: 'rgba(0, 0, 0, 0.75)',
  },
  content: {
    display: 'flex',
    flexFlow: 'column nowrap',
    justifyContent: 'space-between',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    marginTop: 'auto',
    marginRight: 'auto',
    marginBottom: 'auto',
    marginLeft: 'auto',
    width: 500,
    height: 400,
    maxHeight: 400,
    overflow: 'visible',
  },
}

const actionStyle = {
  display: 'flex',
  flexFlow: 'row nowrap',
  justifyContent: 'flex-end',
  marginRight: 12,
}

export class EditEntryDlg extends Component {
  constructor(props) {
    super(props)
    this.okButtonRef = createRef()
  }

  render() {
    const {
      cancelEntryEdit,
      mappings,
      saveEntryEdit,
      shiftGroups: { selectedGroupIdx },
      shiftGroups,
    } = this.props
    const group = shiftGroups.groups[selectedGroupIdx] || {}
    const { editQueue, editing, groupId } = group

    const close = () => cancelEntryEdit(groupId)

    const submit = () => saveEntryEdit(groupId, editQueue)

    if (editing === undefined) {
      return <section />
    }
    return (
      <ReactModal
        isOpen={editing}
        onRequestClose={close}
        style={modalStyle}
      >
        <EntryControls
          group={group}
          {...this.props}
          mappings={mappings}
          okButtonRef={this.okButtonRef}
          onSubmit={submit}
        />

        <div className={instructions}>
          When Group {String.fromCharCode(65 + selectedGroupIdx)} is enabled, this rule will alter any messages that
          match the INPUT so that they take on the characteristics of the OUTPUT.
          <br />
          <br />
          If Note Stack is selected for OUTPUT, then set the stacked note # there.
        </div>

        <div style={actionStyle}>
          <Button
            onClick={close}
            tag="btnCancel"
          >
            Cancel
          </Button>
          <Button
            buttonRef={this.okButtonRef}
            color="primary"
            onClick={submit}
            tabIndex={-1}
            tag="btnApply"
            variant="contained"
          >
            Apply
          </Button>
        </div>
      </ReactModal>
    )
  }
}

EditEntryDlg.propTypes = {
  shiftGroups: groupsShape.isRequired,
  mappings: mappingsShape.isRequired,
  saveEntryEdit: PropTypes.func.isRequired,
  cancelEntryEdit: PropTypes.func.isRequired,
}

export const mapStateToProps = ({ shiftGroups, mappings }) => ({ shiftGroups, mappings })
export const mapDispatchToProps = dispatch => bindActionCreators(
  {
    ...shiftGroupActions, //
    ...shiftEntryActions,
  },
  dispatch,
)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(EditEntryDlg)
