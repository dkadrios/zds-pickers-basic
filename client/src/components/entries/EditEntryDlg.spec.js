import set from 'lodash/fp/set'
import { mappings, shiftGroups } from '../../../__mocks__'
import { EditEntryDlg, mapDispatchToProps, mapStateToProps } from './EditEntryDlg'

describe('EditEntryDlg tests', () => {
  const saveEntryEdit = jest.fn()
  const cancelEntryEdit = jest.fn()
  const changeStatus = jest.fn()
  const changeChannel = jest.fn()
  const changeValue = jest.fn()

  const groupId = 0
  const editQueue = {
    entryId: -1,
    input: { channel: 9, status: 9, value: 0 },
    output: { channel: 9, status: 9, value: 0 },
  }

  const props = {
    shiftGroups,
    mappings,
    saveEntryEdit,
    cancelEntryEdit,
    changeStatus,
    changeChannel,
    changeValue,
  }

  beforeEach(jest.clearAllMocks)

  it('renders correctly', () => {
    shallowExpect(EditEntryDlg)(props).toMatchSnapshot()
  })

  it('renders correctly when there is no group', () => {
    shallowExpect(EditEntryDlg)(set('shiftGroups.selectedGroupIdx', -1)(props)).toMatchSnapshot()
  })

  it('should cancel when button is clicked', () => {
    const wrapper = shallow(<EditEntryDlg {...props} />)
    wrapper
      .find('WithStyles(Button)')
      .filter('[tag="btnCancel"]')
      .simulate('click')
    expect(cancelEntryEdit).toHaveBeenCalledWith(groupId)
  })

  it('should submit when button is clicked', () => {
    const wrapper = shallow(<EditEntryDlg {...props} />)
    wrapper
      .find('WithStyles(Button)')
      .filter('[tag="btnApply"]')
      .simulate('click')
    expect(saveEntryEdit).toHaveBeenCalledWith(groupId, editQueue)
  })

  it('should utilize mapStateToProps', () => {
    const state = { shiftGroups, mappings }
    expect(mapStateToProps(state)).toEqual(state)
  })

  it('should utilize mapDispatchToProps', () => {
    const dispatch = jest.fn()
    mapDispatchToProps(dispatch).cancelEntryEdit(groupId)
    expect(dispatch.mock.calls[0][0]).toEqual({ groupId, type: 'CANCEL_ENTRY_EDIT' })
  })
})
