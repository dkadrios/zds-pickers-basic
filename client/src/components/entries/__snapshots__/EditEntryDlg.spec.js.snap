// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditEntryDlg tests renders correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": 0,
        "display": "flex",
        "flexFlow": "column nowrap",
        "height": 400,
        "justifyContent": "space-between",
        "left": 0,
        "marginBottom": "auto",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "auto",
        "maxHeight": 400,
        "overflow": "visible",
        "right": 0,
        "top": 0,
        "width": 500,
      },
      "overlay": Object {
        "backgroundColor": "rgba(0, 0, 0, 0.75)",
      },
    }
  }
>
  <EntryControls
    cancelEntryEdit={[MockFunction]}
    changeChannel={[MockFunction]}
    changeStatus={[MockFunction]}
    changeValue={[MockFunction]}
    group={
      Object {
        "active": false,
        "channel": 9,
        "editQueue": Object {
          "entryId": -1,
          "input": Object {
            "channel": 9,
            "status": 9,
            "value": 0,
          },
          "output": Object {
            "channel": 9,
            "status": 9,
            "value": 0,
          },
        },
        "editing": false,
        "entries": Array [
          Object {
            "entryId": 0,
            "input": Object {
              "channel": 9,
              "status": 9,
              "value": 1,
            },
            "output": Object {
              "channel": 9,
              "status": 9,
              "value": 1,
            },
          },
          Object {
            "entryId": 1,
            "input": Object {
              "channel": 9,
              "status": 9,
              "value": 2,
            },
            "output": Object {
              "channel": 9,
              "status": 9,
              "value": 2,
            },
          },
          Object {
            "entryId": 2,
            "input": Object {
              "channel": 9,
              "status": 9,
              "value": 3,
            },
            "output": Object {
              "channel": 9,
              "status": 9,
              "value": 3,
            },
          },
          Object {
            "entryId": 3,
            "input": Object {
              "channel": 9,
              "status": 9,
              "value": 4,
            },
            "output": Object {
              "channel": 9,
              "status": 9,
              "value": 4,
            },
          },
        ],
        "groupId": 0,
        "label": "Group A",
        "selectedEntryId": NaN,
        "selectedRows": Array [],
        "sortBy": "SORT_BY_ALL",
        "sortDir": 1,
        "sortOn": 0,
        "value": 110,
      }
    }
    mappings={
      Object {
        "channels": Array [
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          "BFD3",
          null,
          null,
          null,
          null,
          null,
          null,
        ],
        "dialogVisible": false,
        "stockMappings": Array [
          "2box",
          "Addictive Drums",
          "Alesis DM Pro",
          "Alesis DM10",
          "Alesis DM5",
          "BFD2",
          "BFD3",
          "Battery 3",
          "E-MU Protean Acoustic",
          "E-MU Protean Bread n Butter",
          "E-MU Protean Studio",
          "E-MU Protean Synth",
          "EZ Drummer The Classic EZX",
          "EZDrummer",
          "EZDrummer Jazz Kit",
          "EZDrummer Latin Percussion",
          "General MIDI",
          "Infinite Instruments",
          "John Emrich - Articulations 1",
          "Kontakt 2",
          "Ocean Way Drums",
          "Pearl Redbox",
          "Roland HandSonic 15",
          "Roland TD-10",
          "Roland TD-10 Perc 1",
          "Roland TD-10 Perc 2",
          "Roland TD-10 Perc 3",
          "Roland TD-10 Perc 4",
          "Roland TD-12",
          "Roland TD-12 Perc 1 (Latin)",
          "Roland TD-12 Perc 2 (Indian)",
          "Roland TD-12 Perc 3 (African)",
          "Roland TD-12 Perc 4 (Salsa)",
          "Roland TD-12 Perc 5 (SFX)",
          "Roland TD-12 Perc 6-8 (User)",
          "Roland TD-20",
          "Roland TD-20 Perc",
          "Roland TD-3",
          "Roland TD-30",
          "Roland TD-30 Perc 1 (Latin)",
          "Roland TD-30 Perc 2 (Indian)",
          "Roland TD-30 Perc 3 (African)",
          "Roland TD-30 Perc 4 (Salsa)",
          "Roland TD-30 Perc 5 (SFX)",
          "Roland TD-30 Perc 6-8 (User)",
          "Roland TD-5",
          "Roland TD-6",
          "Roland TD-6 Perc 1",
          "Roland TD-6 Perc 10 (Special)",
          "Roland TD-6 Perc 2",
          "Roland TD-6 Perc 3 (Room)",
          "Roland TD-6 Perc 4 (Power)",
          "Roland TD-6 Perc 5 (Elec)",
          "Roland TD-6 Perc 6 (808)",
          "Roland TD-6 Perc 7 (Jazz)",
          "Roland TD-6 Perc 8 (Brush)",
          "Roland TD-6 Perc 9 (Perc)",
          "Roland TD-7",
          "Roland TD-8",
          "Roland TD-8 Perc 1",
          "Roland TD-8 Perc 10 (Special)",
          "Roland TD-8 Perc 2",
          "Roland TD-8 Perc 3 (Room)",
          "Roland TD-8 Perc 4 (Power)",
          "Roland TD-8 Perc 5 (Elec)",
          "Roland TD-8 Perc 6 (808)",
          "Roland TD-8 Perc 7 (Jazz)",
          "Roland TD-8 Perc 8 (Brush)",
          "Roland TD-8 Perc 9 (Perc)",
          "Roland TD-9",
          "Roland TD-9 Perc",
          "Simmons SD1000",
          "Simmons SD5X",
          "Simmons SD7",
          "Simmons SD7PK",
          "Simmons SD9K",
          "Sonic Reality iMap Extended",
          "Sonic Reality iMap Standard",
          "Studio ProFiles",
          "Superior Drummer",
          "Yamaha DTX 700",
          "Yamaha DTXplorer",
          "Zendrum Stompblock",
          "Zendrum TRS",
          "Zendrum TRS 2.0",
        ],
        "userDialogVisible": false,
        "userImportDialogVisible": false,
        "userMappings": Array [
          "BFD3 Custom ZenEdit",
          "reaper",
        ],
      }
    }
    okButtonRef={
      Object {
        "current": null,
      }
    }
    onSubmit={[Function]}
    saveEntryEdit={[MockFunction]}
    shiftGroups={
      Object {
        "groups": Array [
          Object {
            "active": false,
            "channel": 9,
            "editQueue": Object {
              "entryId": -1,
              "input": Object {
                "channel": 9,
                "status": 9,
                "value": 0,
              },
              "output": Object {
                "channel": 9,
                "status": 9,
                "value": 0,
              },
            },
            "editing": false,
            "entries": Array [
              Object {
                "entryId": 0,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 1,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 1,
                },
              },
              Object {
                "entryId": 1,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 2,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 2,
                },
              },
              Object {
                "entryId": 2,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 3,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 3,
                },
              },
              Object {
                "entryId": 3,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 4,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 4,
                },
              },
            ],
            "groupId": 0,
            "label": "Group A",
            "selectedEntryId": NaN,
            "selectedRows": Array [],
            "sortBy": "SORT_BY_ALL",
            "sortDir": 1,
            "sortOn": 0,
            "value": 110,
          },
          Object {
            "active": false,
            "channel": 9,
            "editQueue": Object {
              "entryId": -1,
              "input": Object {
                "channel": 9,
                "status": 9,
                "value": 0,
              },
              "output": Object {
                "channel": 9,
                "status": 9,
                "value": 0,
              },
            },
            "editing": false,
            "entries": Array [
              Object {
                "entryId": 0,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 5,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 5,
                },
              },
              Object {
                "entryId": 1,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 6,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 6,
                },
              },
              Object {
                "entryId": 2,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 7,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 7,
                },
              },
            ],
            "groupId": 1,
            "label": "Group B",
            "selectedEntryId": NaN,
            "selectedRows": Array [],
            "sortBy": "SORT_BY_ALL",
            "sortDir": 1,
            "sortOn": 0,
            "value": 111,
          },
          Object {
            "active": false,
            "channel": 9,
            "editQueue": Object {
              "entryId": -1,
              "input": Object {
                "channel": 9,
                "status": 9,
                "value": 0,
              },
              "output": Object {
                "channel": 9,
                "status": 9,
                "value": 0,
              },
            },
            "editing": false,
            "entries": Array [
              Object {
                "entryId": 0,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 8,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 8,
                },
              },
              Object {
                "entryId": 1,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 9,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 9,
                },
              },
            ],
            "groupId": 2,
            "label": "Group C",
            "selectedEntryId": NaN,
            "selectedRows": Array [],
            "sortBy": "SORT_BY_ALL",
            "sortDir": 1,
            "sortOn": 0,
            "value": 112,
          },
          Object {
            "active": false,
            "channel": 9,
            "editQueue": Object {
              "entryId": -1,
              "input": Object {
                "channel": 9,
                "status": 9,
                "value": 0,
              },
              "output": Object {
                "channel": 9,
                "status": 9,
                "value": 0,
              },
            },
            "editing": false,
            "entries": Array [
              Object {
                "entryId": 0,
                "input": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 10,
                },
                "output": Object {
                  "channel": 9,
                  "status": 9,
                  "value": 10,
                },
              },
            ],
            "groupId": 3,
            "label": "Group D",
            "selectedEntryId": NaN,
            "selectedRows": Array [],
            "sortBy": "SORT_BY_ALL",
            "sortDir": 1,
            "sortOn": 0,
            "value": 113,
          },
        ],
        "maxEntries": 112,
        "selectedGroupIdx": 0,
        "totalEntries": 10,
      }
    }
  />
  <div>
    When Group 
    A
     is enabled, this rule will alter any messages that match the INPUT so that they take on the characteristics of the OUTPUT.
    <br />
    <br />
    If Note Stack is selected for OUTPUT, then set the stacked note # there.
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "flexFlow": "row nowrap",
        "justifyContent": "flex-end",
        "marginRight": 12,
      }
    }
  >
    <WithStyles(Button)
      onClick={[Function]}
      tag="btnCancel"
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      buttonRef={
        Object {
          "current": null,
        }
      }
      color="primary"
      onClick={[Function]}
      tabIndex={-1}
      tag="btnApply"
      variant="contained"
    >
      Apply
    </WithStyles(Button)>
  </div>
</Modal>
`;

exports[`EditEntryDlg tests renders correctly when there is no group 1`] = `<section />`;
